### PR TITLE
Add error msg to help build inputs

### PR DESCRIPTION
**Pull request by ntouran**

### CODE REVIEW

File: armi/reactor/reactors.py [hunk 1]
```diff
@@ -132,6 +132,10 @@ def factory(cs, bp, geom: Optional[geometry.SystemLayoutInput] = None):
 
     with directoryChangers.DirectoryChanger(cs.inputDirectory, dumpOnException=False):
         # always construct the core first (for assembly serial number purposes)
+        if not bp.systemDesigns:
```
Comment (youngmit) [inline armi/reactor/reactors.py:135]: This isn't checking specifically for a "core" system blueprint...

Comment (ntouran) [reply armi/reactor/reactors.py:135]: Directly below it is, so if this branch of the code is being executed, a `core` system is being sought.

File: armi/reactor/components/__init__.py [hunk 2]
```diff
@@ -66,7 +66,15 @@ def factory(shape, bcomps, kwargs):
...
     _removeDimensionNameSpaces(kwargs)
 
-    return class_(components=bcomps, **kwargs)
+    try:
+        return class_(components=bcomps, **kwargs)
+    except TypeError:
+        # TypeError raised when kwarg is missing. We add extra information
+        # to the error to indicate which component needs updating.
+        runLog.error(f"Potentially invalid kwargs {kwargs} for {class_} of shape {shape}."
+                " Check input.")
```
Comment (youngmit) [inline armi/reactor/components/__init__.py:75]: nice! Test might be good tho.

File: armi/reactor/components/basicShapes.py [hunk 3]
```diff
@@ -173,9 +173,9 @@ def __init__(
         Tinput,
         Thot,
         lengthOuter=None,
-        lengthInner=None,
+        lengthInner=0.0,
         widthOuter=None,
-        widthInner=None,
+        widthInner=0.0,
         mult=None,
```
Comment (youngmit) [inline armi/reactor/components/basicShapes.py:179]: Probably wise

**Pull request merged**